pkg/server/event_handler.go

```diff
@@ -293,6 +305,46 @@ func (h *EventHandler[T, P]) getGCSObjectProto(ctx context.Context, objAttrs map
 	return p, nil
 }
 
+type payloadMetadata struct {
+	Metadata map[string]string
+}
+
+func parseGitHubSource(ctx context.Context, data []byte) (*v1alpha1.GitHubSource, error) {
+	pm := &payloadMetadata{}
+	if err := json.Unmarshal(data, pm); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal payloadMetadata %w", err)
+	}
+
+	r := &v1alpha1.GitHubSource{}
+
+	c, found := pm.Metadata["git-commit"]
+	if found {
+		r.Commit = c
```

Comment:
What we do if these are not found?

Reply:
My idea is: if these are not found, than it means the user didn't provide these metadata. So we just ignore that.